### PR TITLE
Use yoda conditions, make private some variables and fix standard code

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug/Hotfix?   | no
| Refactoring?  | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes

Use yoda conditions, fix some line for coding standard and make private some variables

### CODE REVIEW

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 1]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+        $queryBuilderFactory->setOrFilters($fields);
+
+        $this->assertAttributeEquals($fields, 'orFiltering', $queryBuilderFactory);
+    }
+
+    public function testGetOrFilters()
+    {
+        $fields = ['id'];
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setOrFilters($fields);
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: raname $fields to $filters

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 2]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+    {
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->getFields();
+    }
+
+    public function testSetOrFilters()
+    {
+        $fields = ['id'];
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setOrFilters($fields);
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: raname $fields to $filters

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 3]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+        $fieldsReturned = $queryBuilderFactory->getOrFilters();
+
+        $this->assertEquals($fields, $fieldsReturned);
+    }
+
+    public function testSetSorting()
+    {
+        $fields = ['id'];
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setSorting($fields);
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: raname $fields to something related to ... sorting.

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 4]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+
+        $this->assertAttributeEquals($fields, 'sorting', $queryBuilderFactory);
+    }
+
+    public function testGetFilters()
+    {
+        $fields = ['id'];
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setFilters($fields);
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: raname $fields to $filters

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 5]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+        $fieldsReturned = $queryBuilderFactory->getFilters();
+
+        $this->assertEquals($fields, $fieldsReturned);
+    }
+
+    /**
+     * @expectedException \RuntimeException
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: Add a custom Exception in Mado\QueryBundle\Exceptions\ folder.

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 6]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+        $qb = $queryBuilderFactory->getQueryBuilder();
+
+        $this->assertInstanceOf('Doctrine\ORM\QueryBuilder', $qb);
+    }
+
+    public function testGetRel()
+    {
+        $rel = 'foo';
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setRel($rel);
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: 👍

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 7]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setPrinting($print);
+        $printReturned = $queryBuilderFactory->getPrinting();
+
+        $this->assertEquals($print, $printReturned);
+    }
+
+    public function testSetPage()
+    {
+        $page = 'foo';
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: page should be a number, .. not a string. Maybe we can add some type hinting.

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 8]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+        $queryBuilderFactory->setPage($page);
+
+        $this->assertAttributeEquals($page, 'page', $queryBuilderFactory);
+    }
+
+    public function testGetPage()
+    {
+        $page = 'foo';
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setPage($page);
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: as above, ... we should start to add some type hinting also here

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 9]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+        $pageReturned = $queryBuilderFactory->getPage();
+
+        $this->assertEquals($page, $pageReturned);
+    }
+
+    public function testSetPageLength()
+    {
+        $pageLength = 100;
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: 👍

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 10]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setPageLength($pageLength);
+
+        $this->assertAttributeEquals($pageLength, 'pageLength', $queryBuilderFactory);
+    }
+
+    public function testGetPageLength()
+    {
+        $pageLength = 100;
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: 👍

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 11]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+        $queryBuilderFactory->setSelect($select);
+        $selectReturned = $queryBuilderFactory->getSelect();
+
+        $this->assertEquals($select, $selectReturned);
+    }
+
+    public function testGetEntityManager()
+    {
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $emReturned = $queryBuilderFactory->getEntityManager();
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: $returnedEm

File: src/Mado/QueryBundle/Queries/AbstractQuery.php [hunk 12]
```diff
@@ -15,6 +15,8 @@ class AbstractQuery
 
     protected $parser;
 
+    protected $qBuilder;
```
Comment (sensorario) [inline src/Mado/QueryBundle/Queries/AbstractQuery.php]: 👍 👍

File: src/Mado/QueryBundle/Queries/QueryBuilderFactory.php [hunk 13]
```diff
@@ -319,12 +318,12 @@ private function applyFilterOr(
             $salt = '';
             foreach ($this->qBuilder->getParameters() as $parameter) {
                 if ($parameter->getName() == 'field_' . $filterObject->getFieldName()) {
-                    $salt = '_' . rand(111, 999);
+                    $salt = '_' . random_int(111, 999);
```
Comment (sensorario) [inline src/Mado/QueryBundle/Queries/QueryBuilderFactory.php]: 👍

File: tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php [hunk 14]
```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
...
+    /**
+     * @expectedException \RuntimeException
+     */
+    public function testGetQueryBuilderThrowExceptionIfNull()
+    {
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->getQueryBuilder();
+    }
+
+    public function testGetQueryBuilder()
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php]: this test isnt necessary: to assert that returned object is of a specific type we just need php's type hinting.

**Comment by sensorario** [comment]
Change base to 2.2.

**Comment by AlessandroMinoccheri** [comment]
This PR needs to start from 2.2 not from master

**Pull request closed without merging**